pinot-common/src/main/java/org/apache/pinot/common/function/scalar/IpAddressFunctions.java

```diff
@@ -0,0 +1,194 @@
+    IPAddressString prefix = new IPAddressString(prefixStr);
+    IPAddress prefixAddr;
+    try {
+      prefixAddr = prefix.toAddress();
+    } catch (AddressStringException e) {
+      throw new IllegalArgumentException("Invalid IP Address format for " + prefix);
+    }
+    if (!prefixAddr.isPrefixed()) {
+      throw new IllegalArgumentException("IP Address " + prefixStr + " should be prefixed.");
+    }
+    try {
+      return prefixAddr.toPrefixBlock();
+    } catch (PrefixLenException e) {
+      throw e;
+    }
+  }
+
+  /**
+   * Validates IP address ipString and returns IPAddress if validated
+   */
+  private static IPAddress getAddress(String ipString) {
+    try {
+      return new IPAddressString(ipString).toAddress();
+    } catch (AddressStringException e) {
+      throw new IllegalArgumentException("Invalid IP Address format for " + ipString);
+    }
+  }
+
+  @ScalarFunction
+  public static boolean isSubnetOf(String ipPrefix, String ipAddress) {
+    IPAddress prefix = getPrefix(ipPrefix);
+    IPAddress ip = getAddress(ipAddress);
+    return prefix.contains(ip);
+  }
+
+  // -------------- IN HOUSE IMPLEMENTATION STARTS HERE --------------
+  private static String[] fromPrefixToPair(String ipPrefix) {
+    if (!ipPrefix.contains("/")) {
+      throw new IllegalArgumentException("Invalid IP prefix: " + ipPrefix);
+    }
+    return ipPrefix.split("/");
+  }
+
+  private static byte[] fromStringToBytes(String ipAddress) {
+    byte[] address;
+    try {
+      address = InetAddresses.forString(ipAddress).getAddress();
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException("Invalid IP: " + ipAddress + " due to " + e.getMessage());
+    }
+    if (address.length != 4 && address.length != 16) {
+      throw new IllegalArgumentException(
+          "Valid IP address should have length 32 (IPv4) or 128 (IPv6). Invalid IP " + ipAddress + " has length "
+              + address.length);
+    }
+    return address;
+  }
+
+  private static byte[] getAddrMin(byte[] addr, int subnetSize) {
+    int numRangeBits = addr.length * 8 - subnetSize;
+    for (int i = 0; i < addr.length; i++) {
+      if (numRangeBits > i * 8) {
+        int shift = (numRangeBits - i * 8) < 8 ? (numRangeBits - i * 8) : 8;
+        addr[addr.length - 1 - i] &= -0x1 << shift;
+      }
+    }
+    return addr;
+  }
+
+  private static byte[] getAddrMax(byte[] addr, int subnetSize) {
+    int numRangeBits = addr.length * 8 - subnetSize;
+    for (int i = 0; i < addr.length; i++) {
+      if (numRangeBits > i * 8) {
+        int shift = (numRangeBits - i * 8) < 8 ? (numRangeBits - i * 8) : 8;
+        addr[addr.length - 1 - i] |= ~(-0x1 << shift);
+      }
+    }
+    return addr;
+  }
+
+  /**
+   * Returns true if ipAddress is in the subnet of ipPrefix (IPv4 or IPv6)
+   */
+  @ScalarFunction
+  public static boolean isSubnetOfV1(String ipPrefix, String ipAddress)
+      throws UnknownHostException {
```

Comment:
why do we need a v1 ? and why is it throwing exceptions?

Reply:
In this PR, there are 2 paths of implementations as [discussed here](https://github.com/apache/pinot/pull/9501#issuecomment-1282966848). 
The v1 is the in-house implementation without calling the library.